aff4/aff4_image.cc

```diff
@@ -23,45 +23,45 @@ specific language governing permissions and limitations under the License.
 namespace aff4 {
 
 
-AFF4Status CompressZlib_(const std::string &input, std::string* output) {
+AFF4Status CompressZlib_(const std::string &input, std::string& output) {
```

Comment:
Please see this for explanation of this convention:
https://google.github.io/styleguide/cppguide.html#Reference_Arguments

Reply:
Oh. You're using the Google C++ Style Guide. I had not realized that. This is one of those points where I fundamentally disagree with it.

Reply:
For me the main benefit of this convention is at the call site. It becomes clear what parameters are getting modified and what are being read:

```
RETURN_IF_ERROR(CompressZlib_(data, c_data));
```

It is unclear that c_data is about to get trashed. But this is more obvious (at least with stack allocated locals):
```
RETURN_IF_ERROR(CompressZlib_(data, &c_data));
```
 Clearly data is being read and c_data is being written - I dont even need to look at the func prototype.

The only advantage going for the other way is that you cant pass null pointers. Which is nice but its never actually a problem since at the call site we always use &c_data.

Reply:
I don't see the benefits and think it makes the code more confusing, not less. But I'll change these back, as it's not my project.

Reply:
Done. I also tossed in the simple version of the move assignment operator for `AFF4ScopedPtr`.

Reply:
I agree with @uckelman-sf on this one.  Obviously, you should follow your style guidelines if you want to be pedantic about it, but this is a dumb rule as written, IMHO.  Their reasoning of "references are confusing" is silly and the rest of that rule (where they recommend const references over pointers) contradicts that.  Using pointers over references for something like this means that you have to be damn certain you're not ever passing it null, and the function prototype doesn't easily tell me if nullptr is valid or not (in the case of an optional parameter vs required pointer parameter).